### PR TITLE
Added basic_operations.py from pyFV3/stencils to ndsl/stencils

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
Added basic_operations.py from pyFV3/stencils to ndsl/stencils

Fixes # [18](https://github.com/NOAA-GFDL/NDSL/issues/18)

**How Has This Been Tested?**
Tested via lint and currently implemented translate tests
OS: RHEL 8.9 Ootpa
Built conda environment using workflow in github action configurations

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
This needs unit tests

File: tests/test_basic_operations.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    testfield = full(
+        backend=backend,
+        dtype=Float,
+        shape=(nx + 2 * nhalo, ny + 2 * nhalo, nz),
+        fill_value=2.0,
+    )
+
+    setvalue(f_out=outfield, value=2.0)
+
+    assert outfield.any() == testfield.any()
```
Comment (FlorianDeconinck) [inline tests/test_basic_operations.py:184]: `assert (outfield == testfield).all()` ?

File: tests/test_basic_operations.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        factor: FloatField,
+        f_out: FloatField,
+    ):
+        self._adjust_divide_stencil(factor, f_out)
+
+
+def test_copy():
+    copy = Copy(stencil_factory)
+
+    infield = ones(
```
Comment (FlorianDeconinck) [inline tests/test_basic_operations.py:125]: We should probably use `Quantity` to make those more feature full if we need anything, but that's a detail.

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
Doc all stencils and we should be gone

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]

File: tests/test_basic_operations.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    outfield = Quantity(
+        data=np.ones([20, 20, 79]),
+        dims=[X_DIM, Y_DIM, Z_DIM],
+        units="m",
+    )
+
+    copy(f_in=infield.data, f_out=outfield.data)
+
+    assert (infield.data == outfield.data).any()
```
Comment (FlorianDeconinck) [inline tests/test_basic_operations.py:141]: `.all()` since we want all of them to match

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]

File: tests/test_basic_operations.py (hunk 3 above)
Comment (fmalatino) [reply tests/test_basic_operations.py:141]: Ah yes, my bad

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

**Pull request merged**